cli/tools/bundle/html.rs

```
@@ -0,0 +1,358 @@
+
+    Element {
+      attributes,
+      children: vec![],
+      content: None,
+      span: Span::default(),
+      tag_name: atom!("script"),
+      namespace: Namespace::HTML,
+      is_self_closing: false,
+    }
+  }
+}
+
+pub trait VisitHtml: Sized {
+  fn visit_element(&mut self, e: &mut Element) {
+    walk::element(e, self);
+  }
+  fn visit_text(&mut self, _t: &mut Text) {}
+  fn visit_comment(&mut self, _c: &mut Comment) {}
+  fn visit_document_type(&mut self, _d: &mut DocumentType) {}
+  fn visit_document(&mut self, d: &mut Document) {
+    for child in &mut d.children {
+      match child {
+        Child::Element(e) => self.visit_element(e),
+        Child::Text(t) => self.visit_text(t),
+        Child::Comment(c) => self.visit_comment(c),
+        Child::DocumentType(d) => self.visit_document_type(d),
+      }
+    }
+  }
+}
+
+pub mod walk {
+  use swc_html::ast::Element;
+
+  use super::VisitHtml;
+
+  pub fn element(elem: &mut Element, visitor: &mut impl VisitHtml) {
+    for e in &mut elem.children {
+      match e {
+        swc_html::ast::Child::DocumentType(document_type) => {
+          visitor.visit_document_type(document_type)
+        }
+        swc_html::ast::Child::Element(element) => {
+          visitor.visit_element(element)
+        }
+        swc_html::ast::Child::Text(text) => visitor.visit_text(text),
+        swc_html::ast::Child::Comment(comment) => {
+          visitor.visit_comment(comment)
+        }
+      }
+    }
+  }
+}
+
+struct Visitor {
+  scripts: Vec<Script>,
+}
+
+fn get_attr<'a>(e: &'a Element, name: &str) -> Option<&'a str> {
+  e.attributes
+    .iter()
+    .find(|a| a.name == name)
+    .and_then(|a| a.value.as_ref().map(|v| v.as_str()))
+}
+
+impl VisitHtml for Visitor {
+  fn visit_element(&mut self, e: &mut Element) {
+    if e.tag_name == "script" {
+      let src = get_attr(e, "src");
+      let typ = get_attr(e, "type");
+      if let (Some(src), Some(typ @ "module")) = (src, typ) {
```

So we're gonna be silently ignoring `non-module` scripts? That feels odd... Is there a problem bundling scripts?